### PR TITLE
Case 21676: Fix wearable duplication on domain switch (RC81)

### PR DESCRIPTION
**Pull request by SamGondelman**
https://highfidelity.manuscript.com/f/cases/21676/RC81-Wearables-clone-themselves-when-switching-domains-and-the-clones-persist-even-after-the-user-has-removed-the-original

Test plan: https://github.com/highfidelity/hifi/pull/15187

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18145/HighFidelity-Beta-PR15188-c026710.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18145/HighFidelity-Beta-Interface-PR15188-c026710.dmg

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18145/HighFidelity-Beta-PR15188-c026710.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18145/HighFidelity-Quest-Beta-PR15188-c026710.apk)

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18145/HighFidelity-Beta-PR15188-c026710.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18145/HighFidelity-Beta-Interface-PR15188-c026710.exe

File: interface/src/Application.cpp [hunk 1]
```diff
@@ -6947,9 +6947,6 @@ void Application::updateWindowTitle() const {
 }
 
 void Application::clearDomainOctreeDetails(bool clearAll) {
-    // before we delete all entities get MyAvatar's AvatarEntityData ready
-    getMyAvatar()->prepareAvatarEntityDataForReload();
-
```
Comment (AndrewMeadows) [inline interface/src/Application.cpp]: I believe this code will introduce a bug where some recently changed AvatarEntityData state (within the last 10 seconds) might be lost on **quit**.  I recommend adding a `saveAvatarEntityDataToSettings()` call to the `if (_aboutToQuit)` logic block below.

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18193/HighFidelity-Beta-PR15188-bf360d1.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18193/HighFidelity-Beta-Interface-PR15188-bf360d1.dmg

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18193/HighFidelity-Beta-PR15188-bf360d1.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18193/HighFidelity-Quest-Beta-PR15188-bf360d1.apk)

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18193/HighFidelity-Beta-PR15188-bf360d1.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18193/HighFidelity-Beta-Interface-PR15188-bf360d1.exe

**Pull request merged**